data/src/main/java/com/microsoft/azure/kusto/data/ClientRequestProperties.java

```diff
@@ -248,11 +250,47 @@ public static ClientRequestProperties fromString(String json) throws JsonProcess
     }
 
     public String getClientRequestId() {
-        return (String) getOption(OPTION_CLIENT_REQUEST_ID);
+        return _clientRequestId;
     }
 
     public void setClientRequestId(String clientRequestId) {
-        setOption(OPTION_CLIENT_REQUEST_ID, clientRequestId);
+        _clientRequestId = clientRequestId;
+    }
+
+    /**
+     * Gets the application name for tracing purposes.
+     * Overrides the application name set in the connection string.
+     * @return The application name.
+     */
+    public String getApplication() {
+        return _application;
```

Comment:
Are we ignoring the CRP property collection versions of this flag?

Reply:
The server sets the CRP automatically if the headers x-ms-app or x-ms-user are set.

Reply:
Delayed to after discussion